### ISSUE TITLE
Failing to share VPC/Subnets from Network account to new accounts. 

### ISSUE THREAD

**Issue by snemir2** [awslabs/landing-zone-accelerator-on-aws#151]
**Is your feature request related to a problem? Please describe.**
The setup: 
1. Newly deployed v 1.4  aws-best-practices-cccs-medium environment
2. Adding a new Prod account, i.e. in accounts-config.yaml added the following: 
```
  - name: ProdDemo01
    description: Prod account type 
    email: sergey+lza3-ProdDemo01@xxx.com # <----- UPDATE EMAIL ADDRESS
    organizationalUnit: Prod
```
AWSAccelerator-ToolkitProject pipeline step fails as follows:
```
 ❌  AWSAccelerator-NetworkAssociationsStack-690120328019-us-east-2 failed: Error [ValidationError]: Unable to fetch parameters [/accelerator/network/vpc/Prod/id] from parameter store for this account.
    at Request.extractError (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/protocol/query.js:50:29)
    at Request.callListeners (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/codebuild/output/src364/src/s3/00/source/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'ValidationError',
  time: 2023-05-16T05:09:21.347Z,
```

**Describe the feature you'd like**
Would like for pipeline not to fail.  Looks like vpc/subnets actually got shared to the ProdDemo01 account -- but did not get tagged. 
I would want the key/value tags show up in the "shared to" account for easy identification of subnets (public/private, etc)
**Additional context**

**Comment by snemir2** [comment]
fyi: this should be a bug, not a feature

**Comment by snemir2** [comment]
P.S. manually setting `/accelerator/network/vpc/Prod/id` SSM Parameters in the newly created account fixed the issue and allowed pipeline to continue (the parameter existed in the Network account)

**Comment by snemir2** [comment]
Thank you for looking into this.   I do not ... see anything wr
[AWSAccelerator-NetworkVpc-CustomSsmPutParameterVal.tgz](https://github.com/awslabs/landing-zone-accelerator-on-aws/files/11503336/AWSAccelerator-NetworkVpc-CustomSsmPutParameterVal.tgz)
ong in the logs (full logs attached). I had same problem in dev and prod accounts where it could not find the parameters. 

Attaching all logs that i have for the lambda..

**Comment by awsclemj** [comment]
@snemir2 thank you for the logs. Would you mind also providing your `network-config.yaml` so I can try replicating the error on my end? Please also let us know the offending VPC(s) in your configuration file.

Thank you!

**Comment by awsclemj** [comment]
Thank you for reporting this issue, @snemir2. We have identified the root cause and will target a fix to be included in our next release. I will leave this issue open and we will inform you once the fix has been made available. 

Thank you for your interest in Landing Zone Accelerator!

**Comment by snemir2** [comment]
Thank you for a quick response!

**Comment by erik-essextec** [comment]
other than manually copy/pasting the SSM param, are there any other workarounds?  we are now seeing this issue w/v1.4.1  (VPC created prior to v1.4.0...)

**Comment by bo1984** [comment]
Hi Erik, the SSM Parameter store that Jimmy mentioned earlier in this thread is created as part of the custom resource. Without an UPDATE or CREATE invoke event triggering the custom resource (as you mentioned the VPC was pre-existing prior to 1.4.0). The only other methods outside of manually creating the SSM parameter store is to try and recreate the subnets which would trigger this event. 

Important: If you have running workloads in the VPC Subnets, this would not be a recommended path and you should create the parameter paths manually.

**Comment by awsclemj** [comment]
Hello all,

This issue has been resolved in [release/v1.4.2](https://github.com/awslabs/landing-zone-accelerator-on-aws/releases/tag/v1.4.2). Thank you for your support of the solution! I will go ahead and close this issue.
